pkg/client/sharedclientfactory.go

```diff
@@ -188,7 +186,7 @@ func (s *sharedClientFactory) ForResourceKind(gvr schema.GroupVersionResource, k
 	}
 
 	client = NewClient(gvr, kind, namespaced, s.rest, s.timeout)
```

Comment:
Would it make sense to just call `.setConfig` from inside `NewClient`?

Reply:
This was an attempt to maintain compatibility with other consumers of lasso. I figured this approach would be less disruptive since adding it to NewClient would require adding another param which would break anyone that updates lasso.